controllers/etcd_controller.go

```diff
@@ -1519,40 +1510,47 @@ func canDeleteStatefulset(sts *appsv1.StatefulSet, etcd *druidv1alpha1.Etcd) boo
 
 func bootstrapReset(etcd *druidv1alpha1.Etcd) {
 	etcd.Status.Members = nil
-	etcd.Status.ClusterSize = pointer.Int32Ptr(int32(etcd.Spec.Replicas))
+	etcd.Status.ClusterSize = pointer.Int32Ptr(etcd.Spec.Replicas)
 }
 
-func (r *EtcdReconciler) updateEtcdErrorStatus(ctx context.Context, op operationResult, etcd *druidv1alpha1.Etcd, sts *appsv1.StatefulSet, lastError error) error {
+func clusterInBootstrap(etcd *druidv1alpha1.Etcd) bool {
+	return etcd.Status.Replicas == 0 ||
+		(etcd.Spec.Replicas > 1 && etcd.Status.Replicas == 1)
```

Comment:
Why do we need this check? is it for an upgrade case from single node to multi node?

Reply:
Yes, exactly. According to [this comment](https://github.com/gardener/etcd-druid/pull/188#discussion_r644553365) scaling from 1 -> 3 (or similar) is the same as bootstrapping, i.e. we don't plan to add members via the [members API](https://etcd.io/docs/v3.3/op-guide/runtime-configuration/#add-a-new-member).

Reply:
Thanks for clarifying